### PR TITLE
remove reserved key word _binary(#1.1-dev)

### PR DESCRIPTION
**Pull request by YANGGMM**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/MO-Cloud/issues/2342

## What this PR does / why we need it:
remove reserved key word _binary

### CODE REVIEW

**Comment by matrix-meow** [comment]
@YANGGMM Thanks for your contributions!

**Here are review comments for file pkg/sql/parsers/dialect/mysql/mysql_sql.go:**
Based on the provided information, it seems that the pull request is related to fixing a bug mentioned in issue #2342. However, there are no changes provided in the pull request diff for the file `pkg/sql/parsers/dialect/mysql/mysql_sql.go`. This makes it difficult to analyze the changes made and provide feedback.

To provide a thorough review, it is necessary to have the actual changes made in the file. Please ensure that the changes are included in the pull request diff so that a proper analysis can be conducted.

**Here are review comments for file pkg/sql/parsers/dialect/mysql/mysql_sql.y:**
Based on the provided information, the pull request seems to address a bug related to the MySQL SQL parser in the MO-Cloud project. The changes are made to the file `mysql_sql.y` in the `pkg/sql/parsers/dialect/mysql` directory.

Upon reviewing the changes, I have identified a few potential issues and suggestions for improvement:

1. **Unused code removal**: The line `|	UNDERSCORE_BINARY` has been removed from the `non_reserved_keyword` section. It is unclear why this change was made, but if it is no longer needed, it should be removed from the codebase. However, if it is still required, the reason for its removal should be clarified.

2. **Lack of context**: The pull request description and changes do not provide enough context to fully understand the purpose and impact of the changes. It would be helpful to include a more detailed explanation of the bug being fixed and how the changes address it. This will make it easier for reviewers to understand the changes and provide more accurate feedback.

3. **Missing test cases**: It is important to ensure that the changes made to the parser do not introduce any regressions or new bugs. Therefore, it is recommended to include test cases that cover the affected functionality. This will help verify the correctness of the changes and provide a safety net for future modifications.

4. **Consistent formatting**: The changes made to the `mysql_sql.y` file should follow the existing code formatting conventions. In the provided diff, there is an extra space before the removed line. It is important to maintain consistent formatting to improve code readability and maintainability.

5. **Security considerations**: Based on the provided information, there are no apparent security issues in the changes made. However, it is always good practice to review code changes for potential security vulnerabilities, especially when modifying parsers or handling user input. It is recommended to conduct a thorough security review to ensure the changes do not introduce any security risks.

To optimize the changes made in the pull request, the following suggestions can be considered:

1. **Modularization**: If the changes made to the MySQL SQL parser are part of a larger effort to improve the parser or add new features, it may be beneficial to modularize the codebase. Breaking down the parser into smaller, reusable components can improve code maintainability and make it easier to add new functionality in the future.

2. **Code documentation**: Adding inline comments or documentation to explain the purpose and behavior of the code can greatly improve its readability and make it easier for other developers to understand and modify the code in the future.

3. **Code review**: It is always beneficial to have multiple developers review the changes before merging them into the codebase. This can help identify potential issues, provide alternative solutions, and ensure the changes align with the project's coding standards and best practices.

Overall, the pull request addresses a bug in the MySQL SQL parser, but there are some areas that need improvement and clarification. By addressing the identified issues and considering the suggested optimizations, the quality and maintainability of the codebase can be improved.

**Here are review comments for file test/distributed/cases/function/func_cast.test:**
Title: https://github.com/matrixorigin/MO-Cloud/issues/2342 (#1.1-dev)

Body: The pull request is related to issue #2342. The purpose of this pull request is to address the issue mentioned in the linked GitHub issue.

Changes:
- Added a new test case file `func_cast.test`.
- Added SQL statements to create a database, create a table, insert data into the table, select data from the table, and drop the database.

Review:

1. The title of the pull request is clear and provides a link to the related GitHub issue. No issues found.

2. The body of the pull request is brief and provides information about the type of PR and the issue it fixes. No issues found.

3. Changes made in the `func_cast.test` file:
   - Added SQL statements to create a database, create a table, insert data into the table, select data from the table, and drop the database.

   Potential problems and suggestions:
   - Security issue: The SQL statements in the test case file include raw user input (`'测试'`, `'M100.0'`, `'Bool'`, `'True'`, `'管理员'`, `_binary '2023-03-01 14:20:29'`, `'报警测试'`, etc.). This can lead to SQL injection vulnerabilities. To address this, it is recommended to use parameterized queries or prepared statements to sanitize the input and prevent SQL injection attacks.
   - Optimization suggestion: The test case file includes a large number of insert statements. It would be more efficient to use bulk insert statements or a single insert statement with multiple rows to improve performance.

Overall, the pull request addresses the issue mentioned in the GitHub issue. However, it is important to address the security issue related to SQL injection vulnerabilities in the test case file. Additionally, optimizing the insert statements can improve performance.

**Review by iamlinjunhong — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by sukki37 — APPROVED** [review]

**Pull request merged**
